RHEcosystemAppEng/dbaas-operator
# [DBAAS-983] DBaaS Inventory Metrics Changes

## Description
[DBaaS Inventory Metrics Updates](https://issues.redhat.com/browse/DBAAS-983) 
Jira Issue: https://issues.redhat.com/browse/DBAAS-983

## Verification Steps on OpenShift Local

- On fresh Openshift Local instance, in a terminal window login with admin credentials and deploy DBaaS Operator 
```shell
oc apply -f config/samples/installation-for-non-OSD-OCP-environments.yaml
```
- Once the DBaaS Operator is fully installed, expose the Prometheus service and navigate to the prometheus service in the browser
```shell
oc project openshift-dbaas-operator
oc expose svc/prometheus-operated
oc get route
```

- Follow usual process to add provider such as MongoDB Atlas with Org Id, Public Key and Secret. ( Namespace: openshift-dbaas-operator, Database Services -> Database Access )
- Check the prometheus console with dbaas_requests_duration_seconds_bucket, dbaas_requests_duration_seconds_count, dbaas_requests_duration_seconds_sum metrics, with resource=inventory and event=create as filter.
- You may see metric dbaas_requests_error_count in case if there were errors, just in case.
- Delete the Inventory resource by Installed Operators -> OpenShift Database Access Operator -> Provider Account -> (desired inventory resource line) -> Delete DBaaSInventory
- Check the prometheus console with dbaas_requests_duration_seconds_bucket, dbaas_requests_duration_seconds_count, dbaas_requests_duration_seconds_sum metrics, with resource=inventory and event=delete as filter.

- Cleanup: Delete the installed operator with
```shell
oc delete -f config/samples/installation-for-non-OSD-OCP-environments.yaml
```

## Type of change


## Checklist (Definition of Done)
